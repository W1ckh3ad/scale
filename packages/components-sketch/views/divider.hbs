<div data-sketch-artboard="Divider" class="artboard">
    <h1 class="title--artboard">Divider</h1>

    <h2 class="title--artboard-section">Horizontal</h2>
    <div class="agent-states--fullwidth">
      <scale-divider></scale-divider>
    </div>

    <h2 class="title--artboard-section">Vertical</h2>
    <div class="agent-states--fullwidth vertical">
      <scale-divider vertical></scale-divider>
    </div>
</div>
