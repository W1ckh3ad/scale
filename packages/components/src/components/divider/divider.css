:host {
  --width: 100%;
  --height: 100%;
  --spacing: var(--scl-spacing-12);
  --color: var(--scl-color-grey-20);
  --border-width: var(--scl-spacing-1);
  --min-height-vertical: var(--scl-spacing-24);

  width: var(--width);
  height: var(--height);
}

.divider {
  padding: var(--spacing);
}

.divider--vertical {
  display: inline-flex;
  height: inherit;
}

.divider__horizontal {
  margin: 0;
  border: 0;
  border-top: var(--border-width) solid var(--color);
}

.divider__vertical {
  display: inline-flex;
  height: inherit;
  min-height: var(--min-height-vertical);
  border-left: var(--border-width) solid var(--color);
}
