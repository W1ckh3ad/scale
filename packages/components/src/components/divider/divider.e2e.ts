import { newE2EPage } from '@stencil/core/testing';

describe('scale-divider', () => {
  it('renders', async () => {
    const page = await newE2EPage();
    await page.setContent('<scale-divider/>');
    const element = await page.find('scale-divider');
    expect(element).toHaveClass('hydrated');
  });
});
