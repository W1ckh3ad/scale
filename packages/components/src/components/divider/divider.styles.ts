import { JssStyle } from 'jss';

export const styles: JssStyle = {
  ':host': {
    width: '100%',
    height: '100%',
  },
  divider: {
    '& hr': {
      border: '0',
      borderTop: '1px solid #dfdfdf',
    },
  },
  divider__vertical: {
    borderLeft: '1px solid #dfdfdf',
    minHeight: '1rem',
  },
  'divider--vertical': {
    '&, divider__vertical': {
      height: 'inherit',
      display: 'inline-flex',
    },
  },
};
