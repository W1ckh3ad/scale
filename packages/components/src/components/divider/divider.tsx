import { Component, Prop, h, Host } from '@stencil/core';
import classNames from 'classnames';
import Base from '../../utils/base-interface';

const name = 'divider';
@Component({
  tag: 'scale-divider',
  styleUrl: './divider.css',
  shadow: true,
})
export class Divider implements Base {
  /** (optional) Divider size */
  @Prop() size?: string = '';
  /** (optional) Divider vertical */
  @Prop() vertical?: boolean = false;

  /** (optional) Injected CSS styles */
  @Prop() styles?: string;

  componentWillUpdate() {}
  disconnectedCallback() {}

  render() {
    return (
      <Host>
        {this.styles && <style>{this.styles}</style>}

        <div class={this.getCssClassMap()} aria-hidden>
          {!this.vertical ? <hr /> : <span class={`${name}__vertical`} />}
        </div>
      </Host>
    );
  }

  getCssClassMap() {
    return classNames(
      name,
      this.size && `${name}--size-${this.size}`,
      this.vertical && `${name}--vertical`
    );
  }
}
