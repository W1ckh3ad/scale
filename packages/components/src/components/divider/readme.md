# scale-divider



<!-- Auto Generated Below -->


## Properties

| Property   | Attribute  | Description                    | Type      | Default     |
| ---------- | ---------- | ------------------------------ | --------- | ----------- |
| `size`     | `size`     | (optional) Divider size        | `string`  | `''`        |
| `styles`   | `styles`   | (optional) Injected CSS styles | `string`  | `undefined` |
| `vertical` | `vertical` | (optional) Divider vertical    | `boolean` | `false`     |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
