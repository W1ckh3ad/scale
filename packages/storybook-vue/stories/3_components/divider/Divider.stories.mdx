import { Meta, Story, ArgsTable, Canvas, Description } from '@storybook/addon-docs/blocks';
import ScaleDivider from "./ScaleDivider.vue";
export const ICON = 'M16.471,8.321 L12.613594,0.502886112 L8.755,8.321 L0.126435258,9.57533801 L6.37,15.66 L4.89610548,24.2548735 L12.613,20.197 L20.3310826,24.2548735 L18.856,15.66 L25.1007528,9.57533801 L16.471,8.321 Z M9.41891527,9.23556376 L12.613,2.761 L15.8082728,9.23556376 L22.951,10.273 L17.7826928,15.3122039 L19.003,22.427 L12.613594,19.067774 L6.223,22.427 L7.44449521,15.3122039 L2.275,10.273 L9.41891527,9.23556376 Z';

<Meta
  title='Components/Divider'
  component={ScaleDivider}
  argTypes={{}}
/>

export const Template = (args, { argTypes }) => ({
  components: { ScaleDivider },
  props: ScaleDivider.props,
  template: `
    <scale-divider
      :vertical="vertical"
      :styles="styles"
    >
    </scale-divider>
  `
})

# Divider

## Standard

<Canvas withSource="none">
  <Story
    name='Standard'
    args={{}}
  >
    {Template.bind({})}
  </Story>
</Canvas>

<ArgsTable story='Standard' />


## Vertical

<Canvas withSource="none">
  <Story
    name='Vertical'
    args={{
      vertical: true,
      styles: JSON.stringify({'divider--vertical': {height: '150px !important'}})
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

```html
<scale-divider vertical></scale-divider>
```

