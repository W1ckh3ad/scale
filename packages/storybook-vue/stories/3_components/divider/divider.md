# Divider

A divider is a thin line that separates content into groups. This way coherent content is more identifiable to the user.

## Variants

### Horizontal

![Image Name](/assets/3_components/divider/image-20200810102609486.png)

### Vertical

![Image Name](/assets/3_components/divider/image-20200810102617797.png)

## Best Practice

### Subtle

Dividers should be noticeable in a layout, but not jarring.

### Secondary

Dividers should only be used if separating elements through white space is not enough.

### Infrequent

Use dividers sparingly, to create groupings rather than separate items.

## Code
