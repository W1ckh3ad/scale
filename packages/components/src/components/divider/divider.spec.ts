import { newSpecPage } from '@stencil/core/testing';
import { Divider } from './divider';
import { styles } from './divider.styles';
import jss from 'jss';

describe('Divider', () => {
  let element;
  let stylesheet;
  beforeEach(async () => {
    element = new Divider();
    stylesheet = element.stylesheet = jss.createStyleSheet(styles as any);
  });

  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [Divider],
      html: `<scale-divider></scale-divider>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should match snapshot when horizontal false', async () => {
    const page = await newSpecPage({
      components: [Divider],
      html: `<scale-divider vertical=true></scale-divider>`,
    });
    expect(page.root.shadowRoot).toBeTruthy();
    expect(page.root).toMatchSnapshot();
  });

  it('should handle css classes', () => {
    element.customClass = 'custom';
    expect(element.getCssClassMap()).toContain('custom');

    element.size = 'small';
    stylesheet.addRule('divider--size-small', {});
    expect(element.getCssClassMap()).toContain(
      stylesheet.classes['divider--size-small']
    );

    element.vertical = true;
    expect(element.getCssClassMap()).toContain(
      stylesheet.classes['divider--vertical']
    );
  });
});
