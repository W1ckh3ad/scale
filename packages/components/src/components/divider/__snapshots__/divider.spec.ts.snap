// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider should match snapshot 1`] = `
<scale-divider>
  <mock:shadow-root>
    <div aria-hidden="" class="divider">
      <hr>
    </div>
  </mock:shadow-root>
</scale-divider>
`;

exports[`Divider should match snapshot when horizontal false 1`] = `
<scale-divider vertical="true">
  <mock:shadow-root>
    <div aria-hidden="" class="divider divider--vertical">
      <span class="divider__vertical"></span>
    </div>
  </mock:shadow-root>
</scale-divider>
`;
